src/main/java/frc/robot/subsystems/hang/docs/Hang.md

```diff
@@ -0,0 +1,58 @@
+|kPositionConversionFactor|logs the conversion factor of the motor position for the encoders|under setupMotors method|
+|kVelocityConversionFactor|logs the conversion factor of the motor velocity for the encoders|under setupMotors method|
+|kBrake|logs if the motors are braked(wont move) for idle mode|under setupMotors method|
+|Kp, Ki, Kd| logs the p,i,and d for motor soothness|under setupMotors method|
+| --- | --- | ---- |
+|Hang Current Velocity|logs the velocity of the motor|under doSendables method|
+|Hang Current Position|logs the position of the motor|under doSendables method|
+|Hang Target Position|logs the postion that is wanted for the motor to go to|under doSendables method|
+|Hang Error|logs the error of the motor (distance from wanted position)|under doSendables method|
+|Hang Current|logs the current of the motor|under doSendables method|
+|Hang Bus Voltage|logs the voltage of the canbus|under doSendables method|
+|Hang Output|logs the power output of the spark max|under doSendables method|
+| --- | --- | ---- |
+|kHangTolerance|logs the tolerance (the allowed error) of the motor|under isAtPosition method|
+
+## How To Use
+to use the subsytem, you need to create a mountable motor connected to a sparkmax/encoder
+you would also need to have the frc dashboard to be able to tweak and modify code for desired function.
+### manual mode (why and how to use)
+manual mode is important to testing the hang mechanism, specifically how stable, and sturcturaly sound the hang is.
+to set up you would need to manualy switch the mode in "setHangMode" to the "MANUAL" enum within smart dashboard.
+After this, to move the arm, you would also have to set the values for the "setManualVelocity" as well as the "setTargetPosition" methods.
+### -----------------------------------------------------------
+### Pid mode (why and how to use)
+Pid mode is used in competition primarily, the reason is that "PID" control is used to smooth the motor movement to not damage the hang mechanism/motor, and to make it more accurate. to set up you would need to manualy switch the mode in "setHangMode" to the "PID" enum within smart dashboard. After this, to move the arm, you would also have to set the values for the "setManualVelocity" as well as the "setTargetPosition" methods. the key to PID mode is to also tune the p, i, d paremeters under the "setPID" method. the easiest way to tune the pid is to start with p, and work down to d, and repeating. p is used to actualy power the motor till it reaches the "SetTargetPosition" value. in very small amounts, increase the p till it starts to jitter around where the set point would be. in most to all cases, I is never used and so for the hang you dont have to worry about tuning it. d is used to dampen movement around the "setTargetPosition" and the motors starting position. very slightly increase d, until there is a little jitter as posibble before repeating.
```

Comment:
you won’t need setManualVelocity for PID mode. Just calling setTargetPosition is enough since you’re only gonna tell the motor where to move to, not the speed to go at (the PID algorithm determines that speed internally so you don’t have to worry about it).

Aside from that, follow the advice I gave up above.